src/transformers/models/iris/modeling_iris.py

```diff
@@ -0,0 +1,1846 @@
+        self.actor_critic = actor_critic
+        self.device = self.actor_critic.conv1.weight.device
+
+    def act(self, obs: torch.FloatTensor, should_sample: bool = True, temperature: float = 1.0) -> torch.LongTensor:
+        input_ac = (
+            obs
+            if self.actor_critic.use_original_obs
+            else torch.clamp(
+                self.discrete_autoencoder.encode_decode(obs, should_preprocess=True, should_postprocess=True)[0], 0, 1
+            )
+        )
+        logits_actions = self.actor_critic(input_ac)[0][0][:, -1] / temperature
+        act_token = Categorical(logits=logits_actions).sample() if should_sample else logits_actions.argmax(dim=-1)
+        return act_token
+
+
+@add_start_docstrings("The IRIS Model", IRIS_START_DOCSTRING)
+class IrisModel(IrisPreTrainedModel):
+    """
+
+    The model,IRIS (Imagination with auto-Regression over an Inner Speech), a data-efficient rl_agent that learns in a world model composed of a
+    discrete autoencoder and an autoregressive Transformer.It learns behaviors by accurately simulating millions of trajectories
+    Refer to the paper for more details: https://arxiv.org/abs/2209.00588
+
+    """
+
+    def __init__(self, config):
+        super().__init__(config)
+
+        self.config = config
+        self.model_device = config.device
+        encoder = IrisEncoder(config)
+        decoder = IrisDecoder(config)
+        discrete_autoencoder = IrisDiscreteAutoEncoder(
+            self.config.vocab_size,
+            self.config.embed_dim_discrete_autoencoder,
+            encoder,
+            decoder,
+        )
+        world_model = IrisWorldModel(
+            obs_vocab_size=discrete_autoencoder.vocab_size,
+            act_vocab_size=self.config.num_actions,
+            config=config,
+        )
+        actor_critic = IrisActorCritic(
+            act_vocab_size=self.config.num_actions,
+            use_original_obs=self.config.use_original_obs_actor_critic,
+        )
+        self.rl_agent = IrisRlAgent(discrete_autoencoder, world_model, actor_critic)
+
+        # Initialize weights and apply final processing
+        self.post_init()
+
+    @add_start_docstrings_to_model_forward(IRIS_INPUTS_DOCSTRING)
+    @replace_return_docstrings(output_type=IrisOutput, config_class=_CONFIG_FOR_DOC)
+    def forward(
+        self,
+        observations: torch.FloatTensor = None,
+        actions: torch.FloatTensor = None,
+        rewards: torch.FloatTensor = None,
+        ends: torch.IntTensor = None,
+        mask_padding: torch.BoolTensor = None,
+        component: str = None,
+        should_preprocess: Optional[bool] = None,
+        should_postprocess: Optional[bool] = None,
+        output_hidden_states: Optional[bool] = True,
+        output_attentions: Optional[bool] = True,
+        return_dict: Optional[bool] = True,
+    ) -> Union[Tuple[torch.FloatTensor], IrisOutput]:
+        r"""
+        Returns:
+
+        Examples:
+
+        ```python
```

Comment:
The model should folow the same pattern as in the decision transformer i.e. 

```
        >>> # forward pass
        >>> with torch.no_grad():
        ...     state_preds, action_preds, return_preds = model(
        ...         states=states,
        ...         actions=actions,
        ...         rewards=rewards,
        ...         returns_to_go=target_return,
        ...         timesteps=timesteps,
        ...         attention_mask=attention_mask,
        ...         return_dict=False,
        ...     )
```

specifically it looks like some elements should be encapsulated into the state